python/cudf/cudf/dataframe/columnops.py

```diff
@@ -219,29 +219,63 @@ def column_select_by_position(column, positions):
     )
 
 
-def build_column(buffer, dtype, mask=None, categories=None, name=None):
+def build_column(
+    buffer, dtype, mask=None, categories=None, name=None, null_count=None
+):
     from cudf.dataframe import numerical, categorical, datetime, string
 
-    if pd.api.types.is_categorical_dtype(dtype):
+    if not (
+        hasattr(dtype, "categories")
+        or hasattr(dtype, "cat")
+        or str(dtype) in ("category", "categorical")
+    ):
```

Comment:
I don't like this chunk of code, instead could we use `pd.api.types.pandas_dtype(...)` instead of `np.dtype(...)`? Wondering if that has similar perf impact?

Reply:
@kkraus14, made those changes after discussing perf impacts offline.